elastic/beats
# [Filebeat] [AWS] Add support to source AWS cloudwatch logs from linked accounts

## Proposed commit message

PR adds support to Cloudwatch logs from source linked accounts. This is implemented by using existing configuration `log_group_arn` and mapping it to `LogGroupIdentifier` of FilterLogEvents API [1]

## Checklist


## Note for reviewers
You could review commit by commit for better understanding of the changes 

## How to test this PR locally

This require a linked cloudwatch account. If already has one, then,

- Push logs to a newly created log group OR use an already existing log group in a source account
  - Note - you may use data-gen Go program to generate and push logs to your log group (using  output `CLOUDWATCH_LOG`) [2] 
- Configure filebeat cloudwatch input with log group ARN to `log_group_arn`
- Run filebeat and observe filebeat logs in Kibana discover

## Related issues

Addresses: https://github.com/elastic/beats/issues/36642
And makes https://github.com/elastic/beats/pull/36645 PR obsolete 
closes https://github.com/elastic/beats/issues/37681

## Next step

Utilize `includeLinkedAccounts` when dealing with prefixes. To be done in a dedicated PR.  


[1] - https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_FilterLogEvents.html
[1] - https://github.com/Kavindu-Dodan/data-gen
